tests/unit/casmutils/structure_tools.cpp

```diff
@@ -0,0 +1,61 @@
+// These are classes that structure depends on
+#include "../../autotools.hh"
+#include <casmutils/definitions.hpp>
+#include <casmutils/misc.hpp>
+#include <casmutils/xtal/coordinate.hpp>
+#include <casmutils/xtal/lattice.hpp>
+#include <casmutils/xtal/site.hpp>
+#include <casmutils/xtal/structure.hpp>
+#include <gtest/gtest.h>
+// This file tests the functions in:
+#include <casmutils/xtal/structure_tools.hpp>
+
+class StructureToolsTest : public testing::Test
+{
+protected:
+    void SetUp() override
+    {
+
+        casmutils::fs::path cubic_path(casmutils::autotools::input_filesdir / "simple_cubic_Ni.vasp");
+        casmutils::fs::path conventional_path(casmutils::autotools::input_filesdir / "conventional_fcc_Ni.vasp");
+        casmutils::fs::path primitive_path(casmutils::autotools::input_filesdir / "primitive_fcc_Ni.vasp");
+        cubic_Ni_struc_ptr.reset(new casmutils::xtal::Structure(casmutils::xtal::Structure::from_poscar(cubic_path)));
```

Comment:
We can make the code more readable with `typedef`/`using`:
```
class StructureToolsTest : public testing::Test
{
using Structure=casmutils::xtal::Structure;
protected:
    void SetUp() override
    {
        //....
        cubic_Ni_struc_ptr.reset(new Structure(Structure::from_poscar(cubic_path)));

        //Alternatively you can use `std::make_unique`, which might be cleaner, especially when constructing
        //with more than one parameter:
        conventional_fcc_Ni_ptr=std::make_unique<Structure>(Structure::from_poscar(conventional_path)));
        //....
    }
    //....
};
```
It's still kind of verbose, but it'll do until we figure out a better way to build the tests.